### PR TITLE
Add role for pruning old HAS resources

### PR DESCRIPTION
**Pull request by johnmcollier**
🪓 

Discussed on slack - HAS seems to be crashing on start up, as it tries to reconcile hundreds of resources at once, Shoubhik suggested clearing out any old HAS CRs that have been left behind, as a one-time measure.

### CODE REVIEW

File: components/authentication/prune-has.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      - delete
+      - update
+    apiGroups:
+      - 'appstudio.redhat.com'
+    resources:
+      - applications
+      - components
+      - componentdetectionqueries
+  - verbs:
+      - create # allows addition of credentials only.
```
Comment (sbose78) [inline components/authentication/prune-has.yaml:18]: throw in `list` ?

Comment (sbose78) [reply components/authentication/prune-has.yaml:18]: oh this is for tekton

Comment (sbose78) [reply components/authentication/prune-has.yaml:18]: nope, secrets. 

You don't need secrets and tekton in the ClusterRole, do you?

**Review by sbose78 — APPROVED** [review]

File: components/authentication/prune-has.yaml (hunk 1 above)
Comment (johnmcollier) [reply components/authentication/prune-has.yaml:18]: Done.

**Comment by johnmcollier** [comment]
Discussed on slack with @sbose78, merging in now.

**Pull request merged**